DI/Services/EmployeeDAL.cs

```diff
@@ -0,0 +1,40 @@
+﻿using DI.Helpers;
+using DI.InterfacesAndAbstractClasses;
+using Newtonsoft.Json;
+namespace DI.Services
+{
+    public class EmployeeDAL : IEmployeeDAL
+    {
+        public Guid AddEmployeeWithManager(EmployeeHasManager employee)
+        {
+            employee.Guid = new Guid();
+            FileHelper.WriteToFile("Employees.txt", JsonConvert.SerializeObject(employee));
+            return employee.Guid;
+        }
+
+        public Guid AddEmployeeWithoutManager(IEmployee employee)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool DeleteEmployee(Guid guid)
+        {
+            string? employees = FileHelper.ReadFile("Employees.txt");
+            if (string.IsNullOrEmpty(employees))
+            {
+                return false;
+            }
+            else
+            {
+                FileHelper.WriteToFile("Employees.txt", JsonConvert.SerializeObject(JsonConvert.DeserializeObject<List<EmployeeHasManager>>(employees).Where(e => !e.Guid.Equals(guid))));
+                return true;
+            }
+
+        }
+
+        public bool UpdateEmployee(Guid guid)
+        {
+            throw new NotImplementedException();
```

Comment:
Please explain why this code is not best practice. and refactor accordingly

Reply:
I just put it as an example for more methods . 
The NotImplementedException is to create methods and add an implementation to them later , its better to return NotImplementedException rather than return a value , thinking that the code is working and the fact its not ; in the other hand giving errors for no reason is bad practice